cypress/integration/tests/developer/reports/download_report.test.ts

```diff
@@ -1,19 +1,124 @@
-        login();
-    });
-
-    beforeEach("Persist session", function () {
-        // Save the session and token cookie for maintaining one login session
-        preservecookies();
-    });
-
-    it("This test is enabling download of HTML and CSV reports", function () {
-        let reportConfig = ReportConfig.getInstance();
-        reportConfig.enableDownloadHtml();
-        reportConfig.enableDownloadCsv();
-    });
-});
+describe(
+    "This test set is enabling HTML/CSV reports download, run analysis of all possible types and downloads report generated for it",
+    { tags: "@tier1" },
+    function () {
+        let mavenCredentials: CredentialsMaven;
+        let sourceControlUsernameCredentials: CredentialsSourceControlUsername;
+        let applicationList = [];
+
+        before("Login and enable download of HTML and CSV reports", function () {
+            // Prevent hook from running, if the tag is excluded from run
+            if (hasToBeSkipped("@tier1")) return;
+
+            login();
+
+            // Enable HTML anc CSV report downloading
+            let reportConfig = ReportConfig.getInstance();
+            reportConfig.enableDownloadHtml();
+            reportConfig.enableDownloadCsv();
+
+            // Creating credentials required for analysis
+            sourceControlUsernameCredentials = new CredentialsSourceControlUsername(
+                data.getRandomCredentialsData(
+                    CredentialType.sourceControl,
+                    UserCredentials.usernamePassword,
+                    true
+                )
+            );
+            sourceControlUsernameCredentials.create();
+
+            mavenCredentials = new CredentialsMaven(
+                getRandomCredentialsData(CredentialType.maven, "None", true)
+            );
+            mavenCredentials.create();
+        });
+
+        beforeEach("Persist session", function () {
+            // Save the session and token cookie for maintaining one login session
+            preservecookies();
+
+            cy.fixture("application").then(function (appData) {
+                this.appData = appData;
+            });
+            cy.fixture("analysis").then(function (analysisData) {
+                this.analysisData = analysisData;
+            });
+        });
+
+        it("Creating source applications", function () {
+            const sourceApplication = new Analysis(
+                getRandomApplicationData("SourceApp", {
+                    sourceData: this.appData[0],
+                }),
+                getRandomAnalysisData(this.analysisData[0])
+            );
+            sourceApplication.create();
+            applicationList.push(sourceApplication);
+            cy.wait(2 * SEC);
+            sourceApplication.analyze();
+            sourceApplication.verifyAnalysisStatus("Completed");
+            sourceApplication.downloadReport("HTML");
+        });
+
+        it("Creating Upload Binary Analysis", function () {
```

Comment:
We can remove this TC and instead add a TC to disable and check if reports cant be downloaded